packages/ast/src/clients/rpc/query/__snapshots__/rpc.query.test.ts.snap

```diff
@@ -0,0 +1,96 @@
+  Accounts(request: QueryAccountsRequest): Promise<QueryAccountsResponse>;
+  /*Accounts returns all the existing accounts
+  
+  Since: cosmos-sdk 0.43*/
+
+  Account(request: QueryAccountRequest): Promise<QueryAccountResponse>;
+  /*Account returns account details based on address.*/
+
+  Params(request: QueryParamsRequest): Promise<QueryParamsResponse>;
+  /*Params queries all parameters.*/
+
+  ModuleAccounts(request: QueryModuleAccountsRequest): Promise<QueryModuleAccountsResponse>;
+  /*ModuleAccounts returns all the existing module accounts.*/
+
+  Bech32Prefix(request: Bech32PrefixRequest): Promise<Bech32PrefixResponse>;
+  /*Bech32 queries bech32Prefix*/
+
+  AddressBytesToString(request: AddressBytesToStringRequest): Promise<AddressBytesToStringResponse>;
+  /*AddressBytesToString converts Account Address bytes to string*/
+
+  AddressStringToBytes(request: AddressStringToBytesRequest): Promise<AddressStringToBytesResponse>;
+  /*AddressStringToBytes converts Address string to bytes*/
+
+}"
+`;
+
+exports[`RPC Query Client 2`] = `
+"export class QueryClientImpl implements Query {
+  private readonly rpc: Rpc;
+
+  constructor(rpc: Rpc) {
+    this.rpc = rpc;
+    this.Accounts = this.Accounts.bind(this);
+    this.Account = this.Account.bind(this);
+    this.Params = this.Params.bind(this);
+    this.ModuleAccounts = this.ModuleAccounts.bind(this);
+    this.Bech32Prefix = this.Bech32Prefix.bind(this);
+    this.AddressBytesToString = this.AddressBytesToString.bind(this);
+    this.AddressStringToBytes = this.AddressStringToBytes.bind(this);
+  }
+
+  Accounts(request: QueryAccountsRequest): Promise<QueryAccountsResponse> {
+    const data = QueryAccountsRequest.encode(request).finish();
+    const promise = this.rpc.request(\\"cosmos.auth.v1beta1.Query\\", \\"Accounts\\", data);
+    return promise.then(data => QueryAccountsResponse.decode(new _m0.Reader(data)));
+  }
+
+  Account(request: QueryAccountRequest): Promise<QueryAccountResponse> {
+    const data = QueryAccountRequest.encode(request).finish();
+    const promise = this.rpc.request(\\"cosmos.auth.v1beta1.Query\\", \\"Account\\", data);
+    return promise.then(data => QueryAccountResponse.decode(new _m0.Reader(data)));
+  }
+
+  Params(request: QueryParamsRequest): Promise<QueryParamsResponse> {
+    const data = QueryParamsRequest.encode(request).finish();
+    const promise = this.rpc.request(\\"cosmos.auth.v1beta1.Query\\", \\"Params\\", data);
+    return promise.then(data => QueryParamsResponse.decode(new _m0.Reader(data)));
+  }
+
+  ModuleAccounts(request: QueryModuleAccountsRequest): Promise<QueryModuleAccountsResponse> {
+    const data = QueryModuleAccountsRequest.encode(request).finish();
+    const promise = this.rpc.request(\\"cosmos.auth.v1beta1.Query\\", \\"ModuleAccounts\\", data);
+    return promise.then(data => QueryModuleAccountsResponse.decode(new _m0.Reader(data)));
+  }
+
+  Bech32Prefix(request: Bech32PrefixRequest): Promise<Bech32PrefixResponse> {
```

Comment:
@assafmo I noticed that the version in cosmjs didn't' have `Bech32Prefix` and `AddressBytesToString` — I'm guessing this is still correct. Does it look ok to you?

Reply:
These queries are new in cosmos-sdk v0.46: https://github.com/cosmos/cosmos-sdk/blob/5019459b1b2028119c6ca1d80714caa7858c2076/proto/cosmos/auth/v1beta1/query.proto#L37-L64

We should probably have a registry of protobuf derived ts definitions for every tagged version of every chain, tendermint, ibc-go, cosmos-sdk, liquidity-staking-module, etc.

Reply:
Then decouple the client code from these definitions. Have the client devs import the definitions of the chain that they're targeting.

Reply:
great catch! ok so this is correct but it's for the 0.46. 

sounds like a good plan to support all the tagged versions. Now you got me thinking... in future we may have a situation where an app wants to use two chains, and one uses 0.45 and the other 0.46 and we need to support both in the same process.

Reply:
an example use case: keplr governance UI and a user needs to vote. One chain uses 0.45 and another 0.46, my guess is, they'd need to have both cosmos versions.